### PR TITLE
Reimplementing MYNN-EDMF as a submodule

### PR DESCRIPTION
**Pull request by joeolson42**
This PR has the following mods:

1.  git submodule add https://github.com/NCAR/MYNN-EDMF.git src/core_atmosphere/physics/MYNN-EDMF
2. did a “git add MYNN-EDMF” in src/core_atmosphere/physics
3. Modified src/core_atmosphere/physics/Makefile to compile MYNN-EDMF submodule
4. Removed old instructions for compiling MYNN-EDMF from the physics_wrf/Makefile
6. in src/core_atmosphere/physics/physics_wrf, I remove old files (that now reside in the submodule directory): “git rm module_mynnedmf.F90; git rm module_mynnedmf_driver.F90; git rm module_mynnedmf_common.F90”

That's all that is needed. It runs same as old scheme.

### CODE REVIEW

**Review by mdtoyNOAA — APPROVED** [review]
A new submodule is born!   :)

**Review by barlage — COMMENTED** [review]
When I'm in a different branch, then checkout this mynn_20241101 branch, the model will not clean because it seems the Makefile is not where the clean expects:

```
make[4]: Leaving directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/UGWP'
( cd MYNN-EDMF; make clean )
make[4]: Entering directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/MYNN-EDMF'
make[4]: *** No rule to make target 'clean'.  Stop.

```

File: src/core_atmosphere/physics/Makefile [hunk 1]
```diff
@@ -266,6 +269,7 @@ clean:
 	( cd physics_noahmp/utility; $(MAKE) clean )
 	( cd TEMPO; $(MAKE) clean )
 	( cd UGWP; $(MAKE) clean )
+	( cd MYNN-EDMF; $(MAKE) clean )
```
Comment (barlage) [inline src/core_atmosphere/physics/Makefile]: or maybe you just need to add the copy command here also; make with AUTOCLEAN also didn't work

Comment (barlage) [reply src/core_atmosphere/physics/Makefile]: it will work only after you have made the model once

**Comment by joeolson42** [comment]
> When I'm in a different branch, then checkout this mynn_20241101 branch, the model will not clean because it seems the Makefile is not where the clean expects:
> 
> ```
> make[4]: Leaving directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/UGWP'
> ( cd MYNN-EDMF; make clean )
> make[4]: Entering directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/MYNN-EDMF'
> make[4]: *** No rule to make target 'clean'.  Stop.
> ```

Is this only before you compile? I think I have seen this too but only when there is nothing to clean, so I didn't take it seriously. After it is compiled, then the clean worked for me.

**Comment by barlage** [comment]
> > When I'm in a different branch, then checkout this mynn_20241101 branch, the model will not clean because it seems the Makefile is not where the clean expects:
> > ```
> > make[4]: Leaving directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/UGWP'
> > ( cd MYNN-EDMF; make clean )
> > make[4]: Entering directory '/mnt/lfs5/BMC/wrfruc/Michael.Barlage/mpas/testing/code/gsl/joe-fork/src/core_atmosphere/physics/MYNN-EDMF'
> > make[4]: *** No rule to make target 'clean'.  Stop.
> > ```
> 
> Is this only before you compile? I think I have seen this too but only when there is nothing to clean, so I didn't take it seriously. After it is compiled, then the clean worked for me.

It's admittedly a special case, but I think someone who uses multiple remotes/branches might encounter this. In this case, I was cleaning a different build without your code after I updated to your branch. I think this can avoided by changing
`( cd MYNN-EDMF; $(MAKE) clean )`
to
`( cd MYNN-EDMF; cp ./MPAS/Makefile .; $(MAKE) clean )`

Up to you if you want to change.

**Comment by barlage** [comment]
@joeolson42 are you expecting answer changes? I'm getting this after one timestep, but I can check my setup if you don't expect changes.

```
nccmp -dsSqf joe-89bf34bb.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc dev-05d4a9123.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc
Variable               Group  Count          Sum      AbsSum          Min          Max       Range         Mean      StdDev
qv                     /      39461   0.00430633  0.00469312 -1.69873e-05  0.000104862  0.00012185  1.09129e-07 1.28031e-06
qc                     /        756  1.89546e-05 0.000118311 -1.63786e-05   1.8309e-05 3.46876e-05  2.50723e-08 1.09919e-06
qr                     /       3121  6.80451e-06 0.000139241 -5.70692e-06  4.83278e-06 1.05397e-05  2.18023e-09 2.81263e-07
qi                     /       1404  4.39073e-11 9.08208e-09 -3.14685e-10  7.08496e-10 1.02318e-09   3.1273e-14 3.11484e-11
qs                     /       3130 -5.20775e-06 2.94001e-05 -5.68823e-06  3.25217e-06  8.9404e-06 -1.66382e-09 1.38469e-07
qg                     /        808 -7.80305e-10  3.6372e-09 -3.73802e-10  3.29237e-10 7.03039e-10 -9.65724e-13 3.01308e-11
ni                     /       1383      9.49856      24.172    -0.723877       8.5625     9.28638   0.00686808    0.238885
nr                     /       3122      78.9697     2806.61     -144.583      93.7224     238.305    0.0252946     5.91045
ng                     /        791    0.0236106    0.420988     -0.06427    0.0714111    0.135681  2.98491e-05     0.00418
nc                     /        721  9.13727e+09  3.6408e+10 -1.59982e+09  1.77587e+09 3.37569e+09   1.2673e+07 2.28868e+08
nifa                   /      33466     -38980.6     52362.2     -4961.71       550.25     5511.96     -1.16478     70.8343
nwfa                   /      36891 -9.15308e+09 3.71083e+10 -1.77186e+09  1.60037e+09 3.37223e+09      -248112 3.23402e+07
volg                   /        797  1.74974e-12 1.43492e-11 -1.46372e-12  3.39551e-12 4.85922e-12  2.19541e-15   1.589e-13
u                      /     165370  -0.00520545     2.69911  -0.00635123   0.00640082   0.0127521 -3.14776e-08   7.121e-05
w                      /      72271   -0.0294897   0.0743693 -5.49778e-05  1.05724e-05 6.55502e-05 -4.08043e-07 1.90173e-06
pressure               /      25281     -91.6433      277.75     -7.47656      1.52344           9  -0.00362499   0.0611082
surface_pressure       /        564      2.59375     12.5625      -0.0625        0.625      0.6875   0.00459885   0.0407383
rho                    /      28910   0.00528758   0.0146536 -0.000115633  0.000301421 0.000417054  1.82898e-07 3.57802e-06
theta                  /      24597     -3.92758     6.43484    -0.160522    0.0276489    0.188171 -0.000159677  0.00190065
relhum                 /      42885      25.8925     77.5167     -0.32164     0.415977    0.737617  0.000603767    0.009487
divergence             /      61240  2.10057e-09 1.63963e-05 -3.06654e-08  3.15267e-08 6.21922e-08  3.43006e-14 7.52721e-10
vorticity              /     107469  1.98531e-09 2.10928e-05 -1.29945e-07  1.29948e-07 2.59893e-07  1.84733e-14 1.60168e-09
ke                     /      58249    -0.355742     6.53925   -0.0162659    0.0198154   0.0360813 -6.10726e-06 0.000367016
uReconstructZonal      /      56762   -0.0488427    0.696636  -0.00377035   0.00291681  0.00668716 -8.60483e-07 5.30231e-05
uReconstructMeridional /      58535  -0.00642788    0.712251  -0.00298738   0.00518036  0.00816774 -1.09813e-07 5.79223e-05
ertel_pv               /      57422       6.1973     7.52159   -0.0816031     0.224768    0.306371  0.000107926  0.00262979
u_pv                   /       1005  0.000829764   0.0169996 -0.000165939  0.000356674 0.000522614  8.25636e-07  3.3492e-05
v_pv                   /       1013  0.000888132   0.0124564 -0.000126839  0.000186443 0.000313282  8.76734e-07 2.07921e-05
theta_pv               /        579   0.00244141   0.0303345 -0.000396729  0.000274658 0.000671387  4.21659e-06 6.52822e-05
vort_pv                /       1017 -2.68089e-09 8.94315e-08 -1.88084e-09  1.80444e-09 3.68527e-09 -2.63608e-12 1.81261e-10
depv_dt_lw             /      54214  3.00157e-06 9.79052e-06 -2.33878e-07  3.16475e-07 5.50353e-07  5.53653e-11 3.56641e-09
depv_dt_sw             /      53679 -1.26595e-08 1.30793e-07 -1.82456e-09  7.25777e-10 2.55034e-09 -2.35837e-13 1.89826e-11
depv_dt_bl             /      56095     0.010168   0.0111936 -0.000112238  0.000306873 0.000419111  1.81264e-07 3.93961e-06
depv_dt_mix            /      66008  3.34746e-10 1.63545e-07 -2.68216e-10   3.7835e-10 6.46565e-10  5.07129e-15 1.23489e-11
dtheta_dt_mp           /       4544  4.64546e-05 0.000464885 -3.45866e-05  7.50224e-06 4.20888e-05  1.02233e-08  6.5781e-07
depv_dt_mp             /      33419 -1.49364e-05 0.000253573 -1.07865e-05  8.62796e-06 1.94145e-05 -4.46942e-10 9.75384e-08
depv_dt_diab           /      58070    0.0101561   0.0113678 -0.000112301  0.000307009 0.000419309  1.74894e-07 3.87392e-06
depv_dt_fric           /      65283 -0.000118069 0.000835313 -3.93651e-05  3.52893e-05 7.46545e-05 -1.80857e-09 3.30738e-07
depv_dt_diab_pv        /       1010  2.61163e-08 6.24994e-07 -3.61154e-08  6.55821e-08 1.01697e-07  2.58577e-11  3.5527e-09
depv_dt_fric_pv        /       1103  8.13598e-09 1.39594e-07 -1.38471e-09  1.70621e-09 3.09092e-09  7.37622e-12 2.26875e-10
rainnc                 /        218  -0.00117705  0.00336923  -0.00106102  0.000683427  0.00174445 -5.39933e-06 9.30226e-05
precipw                /        822     0.222044    0.224387 -0.000705719   0.00410461  0.00481033  0.000270127 0.000395444
hpbl                   /        109    0.0381622   0.0476227  -0.00105286   0.00289917  0.00395203  0.000350112 0.000647055
hfx                    /        781     -391.724     391.724     -7.67429  -0.00168514      7.6726    -0.501567    0.775496
qfx                    /        781  0.000406001 0.000406001  2.52821e-08   5.7098e-06 5.68451e-06  5.19848e-07 5.95154e-07
lh                     /        781         1015        1015    0.0632057      14.2745     14.2113      1.29962     1.48789
q2                     /        740 -0.000589387 0.000589387 -5.82449e-06 -2.32831e-10 5.82426e-06 -7.96469e-07 8.91224e-07
t2m                    /        781     -2.00037     2.00037   -0.0401917 -6.10352e-05   0.0401306  -0.00256129  0.00301103
th2m                   /        781     -2.04919     2.04919   -0.0402222 -6.10352e-05   0.0401611  -0.00262381   0.0030673
skintemp               /        781     -8.29294     8.29294     -0.14209 -0.000396729    0.141693   -0.0106184   0.0118145
sh2o                   /       3217  -0.00165883  0.00187035 -2.10106e-05  5.24521e-06 2.62558e-05 -5.15644e-07 1.43649e-06
smois                  /       4952   -0.0071188  0.00773925 -9.08822e-05  1.09226e-05 0.000101805 -1.43756e-06 4.62644e-06
tslb                   /       2740     -12.6666     12.6666     -0.14209 -3.05176e-05    0.142059  -0.00462286  0.00813667

```

**Comment by barlage** [comment]
Given the LH changes, I wonder if I have Tanya's latest change in one of my executables.

**Comment by barlage** [comment]
Disregard those previous comments. Results are identical:

```
nccmp -dsSqf joe-89bf34bb.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc dev-05d4a9123.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc
Files "joe-89bf34bb.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc" and "dev-05d4a9123.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_18.12.00.nc" are identical.

nccmp -dsSqf joe-89bf34bb.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_19.00.00.nc dev-05d4a9123.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_19.00.00.nc
Files "joe-89bf34bb.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_19.00.00.nc" and "dev-05d4a9123.hrrrv5.gsl.gsl.conus.120km.rap.2024081518/history.2024-08-15_19.00.00.nc" are identical.

```

**Comment by barlage** [comment]
My last request is to add a .gitignore file to the MYNN-EDMF directory with these six lines:

```
*.o
*.mod
*.f90
Makefile
module_mynnedmf_common.F90
module_mynnedmf_driver.F90
```
this will prevent git from claiming there are "untracked changes" in your repo. This will only ignore the copies that are in MYNN-EDMF not the real ones in the MPAS directory.

**Comment by barlage** [comment]
OK, just noticed your new changes.

**Comment by barlage** [comment]
Awesome @joeolson42 , thanks for adding the .gitignore. No more "untracked content"!

**Review by barlage — APPROVED** [review]

**Pull request merged**